packages/core/src/core.ts

```diff
@@ -179,3 +179,15 @@ export default function r2wc<Props extends R2WCBaseProps, Context>(
 
   return ReactWebComponent
 }
+
+export function useImperativeMethods<Methods extends string>(
```

Comment:
whats this for

Reply:
adding methods to the element. so if you call `useImperativeMethods(container, { run: () => console.log('ran') })` inside the component, then when you get the dom reference to the element, you can `element.run()`. I'd rather do this without needing to pass the container, but I can't figure anything out besides wrapping a context around the whole thing. Not a terrible idea tho?

(really, this should be on a different PR and marked with Experimental, but my project needs both these. I'll split it out before removing draft, but still want feedback now.)

Reply:
It's cool! Thinking about using it, It feels a little awkward... I wonder if we could return the container with the typed methods.

```ts
const containerWithMethods = useImperativeContainer(container, { run: () => console.log('run')})

containerWithMethods.run() // defined
containerWithMethods.dontRun() // not defined
```

It should be achievable with a couple of generics and an intersection. Something like:

```ts
export function useImperativeMethods<Container extends HTMLElement | ShadowRoot, Methods extends Record<string, () => unknown>>(
  container:Container,
  methods: Methods
): Container & Methods {
  const element = "host" in container ? container.host : container;

  for (const method in methods) {
    //@ts-ignore
    element[method] = methods[method];
  }

  return element as Container & Methods
}
```
Not sure if you can remove the cast... maybe?

Reply:
Although, if you never intend to use the container in the code, this is all moot

Reply:
This is for using the methods from the outside. ala `document.getElementBy().run()`